File: DekafTestDB/src/core/MssqlTestHelper.java
package org.jetbrains.dekaf.core;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.dekaf.text.Scriptum;
import org.jetbrains.dekaf.text.TextFileFragment;

import java.sql.Connection;
import java.sql.SQLException;
import java.sql.Statement;



/**
 * @author Leonid Bushuev from JetBrains
 **/
public class MssqlTestHelper extends BaseTestHelper<DBFacade> {

  public MssqlTestHelper(@NotNull final DBFacade db) {
    super(db, Scriptum.of(MssqlTestHelper.class));
    getSchemasNotToZap().add("sys");
  }


  /*
  @Override
  public void prepareX1() {
    performCommand(scriptum, "X1");
  }
  */

  @Override
  public void prepareX1() {
    final TextFileFragment x1 = getScriptum().getText("X1");
    getDb().inSessionDo(session -> {
        Connection conn =
          session.getSpecificService(Connection.class,
                                     ImplementationAccessibleService.Names.JDBC_CONNECTION);
        assert conn != null;
        try {
          conn.setAutoCommit(true);
          Statement stmt = conn.createStatement();
          stmt.execute(x1.getText());
          stmt.close();
        }
        catch (SQLException e) {
          throw new RuntimeException(e);
        }
    });
  }




  @Override
  public void prepareX1000() {
    performCommand(getScriptum(), "X10");
    performCommand(getScriptum(), "X1000");
  }

  @Override
  public void prepareX1000000() {
    prepareX1000();
    performCommand(getScriptum(), "X1000000");
  }


}

File: DekafTestDB/src/core/MssqlTestHelper.kt
package org.jetbrains.dekaf.core

import java.sql.Connection
import java.sql.SQLException


/**
 * MS SQL Server test helper.
 *
 * @author Leonid Bushuev
 */
class MssqlTestHelper(db: DBFacade) : BaseTestHelper(db, MssqlTestHelper::class) {

    init {
        schemasNotToZap.add("sys")
    }


    override fun prepareX1() {
        val x1 = scriptum.getText("X1")
        db.inSessionDo { session ->
            val conn = session.getSpecificService(Connection::class.java,
                                                  ImplementationAccessibleService.Names.JDBC_CONNECTION)!!
            try {
                conn.autoCommit = true
                val stmt = conn.createStatement()
                stmt.execute(x1.text)
                stmt.close()
            }
            catch (e: SQLException) {
                throw RuntimeException(e)
            }
        }
    }


    override fun prepareX1000() {
        performCommand(scriptum, "X10")
        performCommand(scriptum, "X1000")
    }

    override fun prepareX1000000() {
        prepareX1000()
        performCommand(scriptum, "X1000000")
    }


}
